jsonschema2pojo-integration-tests/src/test/java/org/jsonschema2pojo/integration/util/JsonSchema2PojoRule.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright ¬© 2010-2014 Nokia
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jsonschema2pojo.integration.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import static org.apache.commons.io.FileUtils.*;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * A JUnit rule that executes JsonSchema2Pojo.
+ *
+ * @author Christian Trimble
+ *
+ */
+public class JsonSchema2PojoRule implements TestRule {
+
+    private File sourcesDir;
+    private File classesDir;
+    private boolean inRule = false;
+    private boolean cleanSourcesDir = true;
+    private boolean cleanClassesDir = true;
```

Comment:
These would be better as `sourceDirInitialized` and 'classesDirInitialized` with initial value false.